### DISCUSSION TITLE
How can we evaluate the records in the visual in power bi using dax studio

### DISCUSSION THREAD

**Discussion by AnindhithaA** [DaxStudio/DaxStudio#723] [Q&A]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by dgosbell** [comment]
I've converted this to a discussion since it's question not a bug or feature request.

To view the dataset that a visual is using in Power BI you can do the following.

1. In Power BI Desktop click on the **View** menu then on **Performance Analyzer**
2. Then in the Performance Analyzer click **Start Recording**
3. When you hover over your visual a new icon called **Analyze this Visual** will appear, click on that
![image](https://user-images.githubusercontent.com/8241830/134750146-e3578ed5-9be5-4b67-b920-845a6906e1c1.png)
4. This will add an entry to the Performance Analyzer pane, expand that and click on the **Copy query** link
![image](https://user-images.githubusercontent.com/8241830/134750191-8e025baf-f479-437c-a764-badbf9beee8c.png)

Now go back to DAX Studio, connect to the same data model, paste in the query and run it.

**Reply by AnindhithaA** [reply]
Thank you for helping me but still I am able to retrieve only top 501 records could you please help me on retrieving all the records in the visual 

Thanks,
Anindhitha

**Reply by dgosbell** [reply]
So that probably means that the visual pages in results on demand and you are just looking at the first page. 

You will probably find a call to the TOPN( ) function in the query restricting the results to the first 501 rows. You could either remove that call to TOPN or just change the 501 to a really large number.
